adapters/engagebdr/engagebdr.go

```diff
@@ -0,0 +1,138 @@
+package engagebdr
+
+import (
+	"encoding/json"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+	//"regexp"
+
+	"fmt"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+)
+
+type EngageBDRAdapter struct {
+	http *adapters.HTTPAdapter
+	URI  string
+}
+
+//func (adapter *EngageBDRAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+func (adapter *EngageBDRAdapter) MakeRequests(request *openrtb.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	errors := make([]error, 0)
+
+	// EngageBDR uses different sspid parameters for banner and video.
+	sspidImps := make(map[string][]openrtb.Imp)
+
+	for _, imp := range request.Imp {
+		var bidderExt adapters.ExtImpBidder
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err != nil {
+			errors = append(errors, &errortypes.BadInput{
+				Message: fmt.Sprintf("Ignoring imp id=%s, error while decoding extImpBidder, err: %s", imp.ID, err),
+			})
+		}
+		impExt := openrtb_ext.ExtImpEngageBDR{}
+		err := json.Unmarshal(bidderExt.Bidder, &impExt)
+		if err != nil {
+			errors = append(errors, &errortypes.BadInput{
+				Message: fmt.Sprintf("Ignoring imp id=%s, error while decoding impExt, err: %s", imp.ID, err),
+			})
+		}
+		if impExt.Sspid == "" {
+			errors = append(errors, &errortypes.BadInput{
+				Message: fmt.Sprintf("Ignoring imp id=%s, no sspid present", imp.ID),
+			})
+		}
+		sspidImps[impExt.Sspid] = append(sspidImps[impExt.Sspid], imp)
+	}
+
```

Comment:
Just a suggestion: Since you're adapter only supports banner or video ads, might be good to add a check somewhere here in this function to discard imps of any other type. I am sure this must be something that you'd be already handling on your end but it's always better to check and fail early.